### PR TITLE
feat: add `blas/base/csrot`

### PR DESCRIPTION
**Pull request by aman-095**
## Description

> What is the purpose of this pull request?

This RFC proposes to add a routine to apply a plane rotation, where the cos and sin (c and s) are real and the vectors cx and cy are single-precision floating-point vectors as defined in [BLAS Level 1](http://www.netlib.org/blas/#_level_1) routines. Specifically adding `@stdlib/blas/base/csrot` is proposed.

## Related Issues

> Does this pull request have any related issues?

None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
I suggest doing a line-by-line port of the changes made in https://github.com/stdlib-js/stdlib/pull/2257.

File: lib/node_modules/@stdlib/blas/base/csrot/lib/ndarray.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		return cy;
+	}
+	viewX = reinterpret( cx, 0 );
+	viewY = reinterpret( cy, 0 );
+	sx = strideX * 2;
+	sy = strideY * 2;
+	ix = offsetX * 2;
+	iy = offsetY * 2;
+	for ( i = 0; i < N; i++ ) {
+		tmp1 = ( c*viewX[ ix ] ) + ( s*viewY[ iy ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/csrot/lib/ndarray.js]: You need to perform float32 emulation here.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/csrot/lib/ndarray.js]: See `srot`.

File: lib/node_modules/@stdlib/blas/base/csrot/lib/csrot.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	var sy;
+	var ix;
+	var iy;
+	var i;
+	var j;
+
+	if ( N <= 0 ) {
+		return cy;
+	}
+	viewX = reinterpret( cx, 0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/csrot/lib/csrot.js]: This can all be refactored to use the `ndarray` API. See `zdrot`.

Comment (aman-095) [reply lib/node_modules/@stdlib/blas/base/csrot/lib/csrot.js]: Yes, I will apply those changes here.

**Comment by aman-095** [comment]
Hey, @kgryte, I have applied the changes based on #2257.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Comment by kgryte** [comment]
@aman-095 Would you mind always cross-referencing to the [tracking issue](https://github.com/stdlib-js/stdlib/issues/2039) in the OP of your PRs? This will help me during review and merge when cross-linking commits to the tracking issue.

**Pull request merged**